pandas/core/base.py

```diff
@@ -738,17 +739,17 @@ def data(self):
     @property
     def itemsize(self):
         """ return the size of the dtype of the item of the underlying data """
-        return self._values.itemsize
+        return self._ndarray_values.itemsize
 
     @property
     def nbytes(self):
         """ return the number of bytes in the underlying data """
-        return self._values.nbytes
+        return self.values.nbytes
```

Comment:
It depends on what we do with `PeriodIndex.values` and `IntervalIndex.values`.

If we change the public `.values` to be an ExtensionArray then there's no difference.

If we leave the public `.values` as an ndarray of object, then we prefer `._values` since it doesn't materialize the ndarray.

So far we've put off that decision, but we'll need to make a choice soon.

Reply:
You have to override this in DatetimeIndex (tz) then for now, this is the reason CI is failing (recursion error)

On the original comment: I think it is better for now to use `._values` instead of `.values` in such cases. `.values` is the public one that might not always be consistent for backwards compatibility, so let's internally use always `_values` to have something certainly consistent? (and if we decide `.values` to return extension arrays, it is just a public wrapper around `._values`, and if we decide not, we don't have to change anything internally)